components/scream/src/share/tests/vertical_interp_tests.cpp

```diff
@@ -0,0 +1,315 @@
+  const auto policy = ESU::get_default_team_policy(3, num_vert_packs);
+  Kokkos::parallel_for("scream_vert_interp_setup_loop", policy,
+     	       KOKKOS_LAMBDA(MemberType const& team) {
+      const int icol = team.league_rank();
+      view_1d<Pack<Real,1>> x1=ekat::subview(p_src, icol);
+      view_1d<Pack<Real,1>> in=ekat::subview(tmp_src, icol);
+      view_1d<Pack<Real,1>> out_1d=ekat::subview(out_1d_test, icol);
+      view_1d<Mask<1>> msk=ekat::subview(mask_1d_test, icol);
+      perform_vertical_interpolation_impl_1d(x1,
+                                             p_tgt,
+                                             in,
+                                             out_1d,
+                                             msk,
+                                             n_layers_src,
+                                             n_layers_tgt,
+                                             icol,
+                                             masked_val,
+                                             team,
+                                             vert_interp);
+  });
+  Kokkos::fence();
+  
+  for(int col=0; col<3; col++){
+    for(int lev=0; lev<4; lev++){
+      REQUIRE(out_1d_test_s(col,lev) == correct_val[col][lev]);
+    }
+  }
+
+  //Check to see if choose different masked value than default that it returns as expected
+  auto out_same = view_2d<Pack<Real,1>>("",3,npacks_tgt);
+  auto out_same_s = Kokkos::create_mirror_view(ekat::scalarize(out_same));
+  auto mask_same = view_2d<Mask<1>>("",3,npacks_tgt);
+  Real mod_mask_val = -999.;
+  perform_vertical_interpolation(p_src,
+				 p_tgt,
+				 tmp_src,
+				 out_same,
+				 mask_same,
+				 n_layers_src,
+				 n_layers_tgt,
+                                 mod_mask_val);
+  correct_val[0][3] = mod_mask_val;
+  correct_val[1][0] = mod_mask_val;
+  correct_val[1][1] = mod_mask_val;
+  correct_val[2][2] = mod_mask_val;
+  correct_val[2][3] = mod_mask_val;
+
+  for(int col=0; col<3; col++){
+    for(int lev=0; lev<4; lev++){
+      REQUIRE(out_same_s(col,lev) == correct_val[col][lev]);
+    }
+  }
+
+}
+
+TEST_CASE("different_level_pack_size_tests"){
+
+  //Testing that get expected answer for nlevs % packsize==0 and nlevs%packsize!=0
+  //and for different pack sizes. 
+  //For this test choose 4 different scenarios:
+  //1) n_layers_src= 2*N,   n_layers_tgt= 2*N
+  //2) n_layers_src= 2*N+1, n_layers_tgt= 2*N
+  //3) n_layers_src= 2*N,   n_layers_tgt= 2*N-1
+  //4) n_layers_src= 2*N+1, n_layers_tgt= 2*N-1
+  //For each scenario target levels are at the midpoint and so should be the average
+  //of the source layers. 
+  //However, the last target level is not at a midpoint. 
+  //For scenario 1) it is at the last source level
+  //For scenario 2) and 3) it is at the 2nd to last source level
+  //For scenario 4) it is at the 3rd to last source level
+  //using namespace scream;
+  //using namespace vinterp;
+
+  const int N = SCREAM_SMALL_PACK_SIZE;
```

Comment:
I have implemented a "run" function, and in cases of pack size =1 it will just run once. In cases of pack size > 1 it will run with pack size of 1 and whatever the pack size is defined to be. I have kept the first TEST_CASE because it tests aspects that are not in this test, for instance the masking and whether calling the 1D interpolation function gives you the same answer. I have updated the first TEST_CASE to have more target levels so that >1 pack will be used, and I also updated the first TEST_CASE to have pack size = SCREAM_PACK_SIZE.